### PR TITLE
Setup hardhat tests

### PR DESCRIPTION
**Pull request by lajarre**
The main takeaway of this is the failing test.

I spread some `[XXX]` around to be discussed (will remove them when finalizing the pr).

Note: this includes a Github action for hardhat.

We can create a new issue to discuss the options for a fix.

### CODE REVIEW

File: src/Minter.sol [hunk 1]
```diff
@@ -8,14 +8,15 @@ interface Balot {
...
+// [XXX] Security: either make it ownable either move everything to the constructor
 contract Minter {
   function safeMintRange(
     address collection,
     address nextOwner,
     address to,
     uint16 start,
     uint16 end,
-    uint16 step
+    uint16 step // [XXX] why not remove this?
```
Comment (TimDaub) [inline src/Minter.sol:19]: yeah, I don't fully see the point of having it either.

File: src/Minter.sol [hunk 2]
```diff
@@ -8,14 +8,15 @@ interface Balot {
   function transferOwnership(address newOwner) external;
 }
 
+// [XXX] Security: either make it ownable either move everything to the constructor
```
Comment (TimDaub) [inline src/Minter.sol:11]: Yeah so do I understand correctly that someone could frontrun us the moment we gave the minter ownership over the NFT collection, right?

**Review by TimDaub — COMMENTED** [review]
Great job! And great find about the frontrunning vector! How do you suggest we proceed with this PR? IMO we can merge despite tests failing now and address #31 separately. OK?

File: src/Minter.sol (hunk 1 above)
Comment (TimDaub) [reply src/Minter.sol:19]: opened #32

**Comment by lajarre** [comment]
> Great job! And great find about the frontrunning vector! How do you suggest we proceed with this PR? IMO we can merge despite tests failing now and address https://github.com/whitecube-online/balot/issues/31 separately. OK?

Let's do this indeed, merging.

**Pull request merged**